cloudnet/src/main/java/de/dytanic/cloudnet/command/commands/CommandMigrate.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019-2021 CloudNetService team & contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.dytanic.cloudnet.command.commands;
+
+import static de.dytanic.cloudnet.command.sub.SubCommandArgumentTypes.anyStringIgnoreCase;
+import static de.dytanic.cloudnet.command.sub.SubCommandArgumentTypes.dynamicString;
+
+import com.google.common.primitives.Ints;
+import de.dytanic.cloudnet.CloudNet;
+import de.dytanic.cloudnet.command.sub.SubCommandBuilder;
+import de.dytanic.cloudnet.command.sub.SubCommandHandler;
+import de.dytanic.cloudnet.common.INameable;
+import de.dytanic.cloudnet.common.concurrent.function.ThrowableConsumer;
+import de.dytanic.cloudnet.common.language.LanguageManager;
+import de.dytanic.cloudnet.database.AbstractDatabaseProvider;
+import de.dytanic.cloudnet.driver.database.Database;
+import java.util.Collection;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+
+public class CommandMigrate extends SubCommandHandler {
+
+  public CommandMigrate() {
+    super(
+      SubCommandBuilder.create()
+        .generateCommand(
+          (subCommand, sender, command, args, commandLine, properties, internalProperties) -> {
+            AbstractDatabaseProvider sourceDatabaseProvider = getDatabaseProvider((String) args.argument(1));
+            AbstractDatabaseProvider targetDatabaseProvider = getDatabaseProvider((String) args.argument(2));
+
+            if (sourceDatabaseProvider.equals(targetDatabaseProvider)) {
+              sender.sendMessage(LanguageManager.getMessage("command-migrate-source-equals-target"));
+              return;
+            }
+
+            executeIfNotCurrentProvider(sourceDatabaseProvider, AbstractDatabaseProvider::init);
+            executeIfNotCurrentProvider(targetDatabaseProvider, AbstractDatabaseProvider::init);
+
+            Integer chunkSize = Ints.tryParse(properties.getOrDefault("chunk-size", Integer.toString(100)));
+            if (chunkSize == null) {
+              chunkSize = 100;
```

Comment:
Telling the user that the input wasn’t a number might be better than just using the default.